functional_tests.go

```diff
@@ -5421,6 +5810,79 @@ func testEncryptedCopyObjectWrapper(c *minio.Client) {
 		logError(testName, function, args, startTime, "", "Downloaded object mismatched for encrypted object", err)
 		return
 	}
+
+	// Test key rotation for source object in-place.
+	dst, err = minio.NewDestinationInfo(bucketName, "srcObject", &key2, nil)
+	if err != nil {
+		logError(testName, function, args, startTime, "", "NewDestinationInfo failed", err)
+		return
+	}
+	args["destination"] = dst
+
+	err = c.CopyObject(dst, src)
+	if err != nil {
+		logError(testName, function, args, startTime, "", "CopyObject failed", err)
+		return
+	}
+
+	// Get copied object and check if content is equal
+	opts = minio.GetObjectOptions{}
+	for k, v := range key2.GetSSEHeaders() {
+		opts.Set(k, v)
+	}
+	reader, _, err = coreClient.GetObject(bucketName, "srcObject", opts)
+	if err != nil {
+		logError(testName, function, args, startTime, "", "GetObject failed", err)
+		return
+	}
+	defer reader.Close()
+
+	decBytes, err = ioutil.ReadAll(reader)
+	if err != nil {
+		logError(testName, function, args, startTime, "", "ReadAll failed", err)
+		return
+	}
+	if !bytes.Equal(decBytes, buf) {
+		logError(testName, function, args, startTime, "", "Downloaded object mismatched for encrypted object", err)
+		return
+	}
+
+	// Test in-place decryption.
+	dst, err = minio.NewDestinationInfo(bucketName, "srcObject", nil, nil)
+	if err != nil {
+		logError(testName, function, args, startTime, "", "NewDestinationInfo failed", err)
+		return
+	}
+	args["destination"] = dst
+
```

Comment:
`src` is using `key1`, while theSSE-C key for `srcobject` has changed to `key2`. We need a new sourceinfo variable here that uses `key2`

Reply:
actually, we don't the source is not unencrypted.

Reply:
Oh, i see you meant CopyObject, there was a problem in GetObject() as well.